docs/tutorials/Astro/creating-widgets.md

```diff
@@ -0,0 +1,911 @@
+</template>
+</AposCodeBlock>
+
+We aren't going to go through this file in detail. The one thing that you will note is that it is making an API call to the ApostropheCMS server:
+```javascript
+const response = await fetch(`/api/v1/oembed?url=${encodeURIComponent(url)}`);
+```
+If desired, you could create [custom API routes](https://docs.apostrophecms.org/reference/module-api/module-overview.html#restapiroutes-self) for any of your widgets or other web components to query in this manner. We will touch on this in the [Creating Pieces tutorial](/tutorials/Astro/creating-pieces.html#approach-2-custom-api-routes) section.
+
+#### Loading the Web Component
+
+In the Apollo `[...slug].astro` file we are adding the web component script to the `startHead` slot:
+
+<AposCodeBlock>
+
+```astro
+<AposLayout title={aposData.page?.title} {aposData} {bodyClass}>
+  <Fragment slot="startHead">
+    <script src="/scripts/video-widget.js"></script>
+  </Fragment>
+  <!-- ... rest of layout -->
+</AposLayout>
+```
+<template v-slot:caption>
+frontend/src/pages/[...slug].astro
+</template>
+</AposCodeBlock>
+
+### Traditional JavaScript Initialization
+
+For widgets that need more traditional JavaScript initialization, like the `SlideshowWidget`, we need a different approach.
+
+#### Widget Template
+
+<AposCodeBlock>
+
+```astro
+---
+const { widget } = Astro.props;
+const { images, _id } = widget;
+---
+
+<div
+  class="slideshow-widget"
+  data-slideshow
+  data-widget-id={_id}
+>
+  <div class="slideshow-content">
+    {images?.map((image, index) => (
+      <div
+        class:list={[
+          'slide',
+          { active: index === 0 }
+        ]}
+        data-slide={index}
+      >
+        <img src={image._urls.max} alt={image.title || ''} />
+      </div>
+    ))}
+  </div>
+
+  <div class="slideshow-controls">
+    <button class="prev" aria-label="Previous slide">←</button>
+    <button class="next" aria-label="Next slide">→</button>
+  </div>
+</div>
+```
+<template v-slot:caption>
+frontend/src/widgets/SlideshowWidget.astro
+</template>
+</AposCodeBlock>
+
+The template just brings in the widget data and creates simple markup to display and navigate through the slides.
+
+#### JavaScript Initialization
+
+<AposCodeBlock>
+
+```js-astro
+// ... slideshow specific code
+
+const slideshows = new Map();
+
+setTimeout(() => {
+  initSlideshows();
+
+  if (window.apos) {
+    apos.bus.$on('refreshed', initSlideshows);
+    apos.bus.$on('modal-resolved', initSlideshows);
+  }
+}, 300);
+
+function initSlideshows() {
+  slideshows.forEach((slideshow) => slideshow.destroy());
+  slideshows.clear();
+
+  setTimeout(() => {
+    document.querySelectorAll('.slideshow-container').forEach((container) => {
+      const slideshowId = container.dataset.slideshowId;
+      if (slideshowId) {
+        slideshows.set(slideshowId, new Slideshow(container));
+      }
+    });
+  }, 100);
+}
+```
+<template v-slot:caption>
+frontend/src/widgets/SlideshowWidget.astro
```

Comment:
So both of these chunks go in the same file? But there's no `script` tag? Confusing

Reply:
Clarified?